app/service/planning_svc.py

```diff
@@ -87,12 +87,16 @@ def __init__(self, data_svc, utility_svc):
         """
         Replace all variables with facts from the combo to build a single test variant
         """
-        score, rewards = 0, []
+        score, rewards, combo_set_id, combo_link_id = 0, [], set(), set()
         for var in combo:
             score += (score + var[2])
+            combo_set_id.add(var[4])
+            combo_link_id.add(var[5])
             rewards.append(var[3])
             copy_test = copy_test.replace('#{%s}' % var[0], var[1])
             clean_test = clean_test.replace('#{%s}' % var[0], var[1])
+        if len(combo_set_id) == 1 and len(combo_link_id) == 1:
```

Comment:
might be good to pull this out into a named function:
def reward_closeness_...()
this would improve readability and make it more organized when chaining more of these types of score improvements